nens_auth_client/views.py

```diff
@@ -109,19 +112,23 @@ def authorize(request):
     # The RemoteUserBackend finds a local user through a RemoteUser
     user = django_auth.authenticate(request, claims=claims)
 
-    # If nothing was found: only a valid invite warrants a new user association
-    if user is None and INVITE_ID_KEY in request.session:
+    # If nothing was found: only a valid invitation warrants a new user association
+    if user is None and INVITATION_KEY in request.session:
         try:
-            invite = Invite.objects.select_related("user").get(
-                id=request.session[INVITE_ID_KEY], status=Invite.PENDING
+            invitation = Invitation.objects.select_related("user").get(
+                slug=request.session[INVITATION_KEY], status=Invitation.PENDING
             )
-        except Invite.DoesNotExist:
-            raise PermissionDenied("Invalid invite key")
-        if invite.user is not None:
-            user = invite.user  # user was created at or before invite creation
-            users.create_remote_user(user, claims)  # associate permanently
+        except Invitation.DoesNotExist:
+            raise PermissionDenied("Invalid invitation key")
```

Comment:
Het hoeft geen invalid key te zijn. Het kan ook zijn dat de status inmiddels iets anders dan `PENDING` is.

De slug is uniek, dus ik zou die eerst ophalen zonder de extra `status=PENDING` erbij. Dan klopt de foutmelding ook.
Stap twee is dan even checken of de status wel `PENDING` is. Dan weet je zeker dat de foutmelding klopt.

(Inloggen => helpdesk krijgt er vragen over => het moet zo duidelijk mogelijk zijn :-) )

Reply:
Ik heb naar aanleiding van deze opmerking de accept_invitation view iets veranderd. Ook heb ik de error meldingen in beide views opgesplitst in "does not exist" en "cannot be accepted because it has status '...'"

Nu geeft hij al een 404 op het accept_invitation endpoint vóórdat de user is ingelogd. Dat is duidelijker dan dat de user pas na 4 redirects een PermissionDenied krijgt omdat de invitation al die tijd niet geldig was of niet bestaat.

Natuurlijk kunnen er edge cases optreden (hij was eerst geldig, maar na de 4 redirects niet meer) - daarvoor blijven deze PermissionDenieds wel staan in de authorize view.

Beter zo?

Reply:
Goeie om gelijk in het begin al een 404 te geven! Dat verduidelijkt de flow.

Lijkt me prima zo.